### PR TITLE
[SYCL][XPTI] Performance improvements to XPTI streams

### PR DESCRIPTION
**Pull request by tovinkere**
Current instrumentation in SYCL runtime has additional overheads due to argument capture even it is not subscribed to or needed. This patch addresses this issue.

### CODE REVIEW

**Review by KseniyaTikhomirova — APPROVED** [review]
LGTM

File: sycl/plugins/level_zero/tracing.cpp [hunk 1]
```diff
@@ -30,6 +30,8 @@ constexpr int GMinVer = 1;
 #ifdef XPTI_ENABLE_INSTRUMENTATION
 static xpti_td *GCallEvent = nullptr;
 static xpti_td *GDebugEvent = nullptr;
+static uint8_t GCallStreamID = 0;
+static uint8_t GDebugStreamID = 0;
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/tracing.cpp:34]: Is there a data race here for multi-threaded apps?

File: sycl/source/detail/plugin.hpp [hunk 2]
```diff
@@ -28,6 +28,8 @@ namespace detail {
 #ifdef XPTI_ENABLE_INSTRUMENTATION
 extern xpti::trace_event_data_t *GPICallEvent;
 extern xpti::trace_event_data_t *GPIArgCallEvent;
+extern uint8_t PiCallStreamID;
+extern uint8_t PiDebugCallStreamID;
```
Comment (smaslov-intel) [inline sycl/source/detail/plugin.hpp:32]: I don't see where these are being initialized (and again, no race condition?)

File: sycl/plugins/level_zero/tracing.cpp
[the remark this answers is not in the document]
Comment (jandres742) [reply sycl/plugins/level_zero/tracing.cpp:1]: @smaslov-intel  I didn't know about this file. Definitely the better place is on the Adapter. I have opened an issue to follow up on that https://github.com/oneapi-src/unified-runtime/issues/998

[the remark this answers is not in the document]
Comment (tovinkere) [reply sycl/plugins/level_zero/tracing.cpp:1]: @smaslov-intel Yes, for now I will be applying the same patch for adapters/cuda in the adapters branch. Can follow up with @jandres72 for level_zero.

File: sycl/source/detail/plugin.hpp (hunk 2 above)
Comment (tovinkere) [reply sycl/source/detail/plugin.hpp:32]: Shouldn't be as they are initialized in initializePlugins() in pi.cpp, which should be a singleton.

File: sycl/plugins/level_zero/tracing.cpp (hunk 1 above)
Comment (tovinkere) [reply sycl/plugins/level_zero/tracing.cpp:34]: Shouldn't be a data race as the initialization should be done once in piPluginInit() where zeEnableTracing() is called and the only place it is called from.

**Review by smaslov-intel — APPROVED** [review]
LGTM, but please try to add a test that no extra overhead if the streams are not being listened

**Comment by tovinkere** [comment]
> LGTM, but please try to add a test that no extra overhead if the streams are not being listened

Yes, thanks!

**Pull request merged**